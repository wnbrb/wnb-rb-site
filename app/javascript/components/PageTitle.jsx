import React from 'react';
import PropTypes from 'prop-types';
import RubyOutline from 'components/icons/RubyOutline';
import 'stylesheets/page-title';

const PageTitle = ({ text, children }) => {
    return (
        <div className="page-title">
            <RubyOutline className="ruby-outline" />
            <div className="title-container">
                <h1 className="title-text">{text}</h1>
                {children}
            </div>
        </div>
    );
};

PageTitle.propTypes = {
    children: PropTypes.node,
    text: PropTypes.string,
};

export default PageTitle;
