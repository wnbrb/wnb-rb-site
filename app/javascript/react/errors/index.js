export class UnauthorizedError extends Error {
    constructor(message) {
        super(message);
        this.name = 'UnauthorizedError';
    }
}
