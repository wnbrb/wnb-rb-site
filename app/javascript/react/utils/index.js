const MS_IN_DAY = 24 * 60 * 60 * 1000;

export const postedAtString = (createdAtString) => {
    const today = new Date();
    const postedDate = new Date(createdAtString);

    const diff = (+today - +postedDate) / MS_IN_DAY;

    if (diff <= 1) {
        return 'today';
    }

    if (diff < 7) {
        return `${Math.floor(diff)} days ago`;
    }

    if (diff < 30) {
        const weekDiff = Math.floor(diff / 7);
        const weekOrWeeks = weekDiff === 1 ? 'week' : 'weeks';

        return `${weekDiff} ${weekOrWeeks} ago`;
    }

    const monthDiff = Math.floor(diff / 30);
    const monthOrMonths = monthDiff === 1 ? 'month' : 'months';

    return `${monthDiff} ${monthOrMonths} ago`;
};
