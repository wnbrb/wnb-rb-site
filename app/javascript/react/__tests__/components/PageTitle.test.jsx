import React from 'react';
import { render } from '@testing-library/react';
import PageTitle from '../../components/PageTitle';

// Mock the NewLogo component
jest.mock('../../components/icons/NewLogo', () => {
    // eslint-disable-next-line react/prop-types
    function MockNewLogo({ className }) {
        return <div className={`mock-new-logo ${className || ''}`}>Logo</div>;
    }
    return MockNewLogo;
});

describe('PageTitle Component', () => {
    test('renders page title with default props', () => {
        const { container } = render(<PageTitle />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders page title with text prop', () => {
        const { container } = render(<PageTitle text="Welcome to our site" />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders page title with text and altText props', () => {
        const { container } = render(<PageTitle text="Events" altText="Events page title" />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders page title with children content', () => {
        const { container } = render(
            <PageTitle text="About Us">
                <p className="subtitle">Learn more about our community</p>
                <button>Get Started</button>
            </PageTitle>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders page title with all props and children', () => {
        const { container } = render(
            <PageTitle text="Contact" altText="Contact us page">
                <div className="contact-info">
                    <span>Email: info@example.com</span>
                </div>
            </PageTitle>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test('applies correct CSS classes', () => {
        const { container } = render(<PageTitle text="Test" />);

        const pageTitle = container.firstChild;
        expect(pageTitle).toHaveClass('page-title');

        const titleContainer = pageTitle.querySelector('.title-container');
        expect(titleContainer).toBeInTheDocument();

        const titleText = pageTitle.querySelector('.title-text');
        expect(titleText).toBeInTheDocument();
        expect(titleText).toHaveClass('title-text');

        const logo = pageTitle.querySelector('.mock-new-logo');
        expect(logo).toHaveClass('new-logo');
    });

    test('renders title text correctly', () => {
        const titleText = 'My Page Title';
        const { getByText } = render(<PageTitle text={titleText} />);

        expect(getByText(titleText)).toBeInTheDocument();
    });

    test('applies altText to title element', () => {
        const altText = 'Page title alt text';
        const { container } = render(<PageTitle text="Title" altText={altText} />);

        const titleElement = container.querySelector('.title-text');
        expect(titleElement).toHaveAttribute('alt', altText);
    });

    test('renders children content correctly', () => {
        const { getByText } = render(
            <PageTitle text="Title">
                <span>Child content</span>
            </PageTitle>,
        );

        expect(getByText('Child content')).toBeInTheDocument();
    });
});
