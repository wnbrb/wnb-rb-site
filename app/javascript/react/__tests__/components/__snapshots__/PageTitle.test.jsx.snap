// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle Component renders page title with all props and children 1`] = `
<div
  class="page-title"
>
  <div
    class="mock-new-logo new-logo"
  >
    Logo
  </div>
  <div
    class="title-container"
  >
    <h1
      alt="Contact us page"
      class="title-text"
    >
      Contact
    </h1>
    <div
      class="contact-info"
    >
      <span>
        Email: info@example.com
      </span>
    </div>
  </div>
</div>
`;

exports[`PageTitle Component renders page title with children content 1`] = `
<div
  class="page-title"
>
  <div
    class="mock-new-logo new-logo"
  >
    Logo
  </div>
  <div
    class="title-container"
  >
    <h1
      class="title-text"
    >
      About Us
    </h1>
    <p
      class="subtitle"
    >
      Learn more about our community
    </p>
    <button>
      Get Started
    </button>
  </div>
</div>
`;

exports[`PageTitle Component renders page title with default props 1`] = `
<div
  class="page-title"
>
  <div
    class="mock-new-logo new-logo"
  >
    Logo
  </div>
  <div
    class="title-container"
  >
    <h1
      class="title-text"
    />
  </div>
</div>
`;

exports[`PageTitle Component renders page title with text and altText props 1`] = `
<div
  class="page-title"
>
  <div
    class="mock-new-logo new-logo"
  >
    Logo
  </div>
  <div
    class="title-container"
  >
    <h1
      alt="Events page title"
      class="title-text"
    >
      Events
    </h1>
  </div>
</div>
`;

exports[`PageTitle Component renders page title with text prop 1`] = `
<div
  class="page-title"
>
  <div
    class="mock-new-logo new-logo"
  >
    Logo
  </div>
  <div
    class="title-container"
  >
    <h1
      class="title-text"
    >
      Welcome to our site
    </h1>
  </div>
</div>
`;
